server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java

```diff
@@ -166,13 +173,18 @@ private Long searchDeveloperId(DbSession dbSession, ComponentTreeWsRequest wsReq
         new LinkedHashSet<>(metricKeys),
         new LinkedHashSet<>(foundMetricKeys));
 
-      throw new NotFoundException(format("The following metric keys are not found: %s", Joiner.on(", ").join(missingMetricKeys)));
+      throw new NotFoundException(format("The following metric keys are not found: %s", COMA_JOINER.join(missingMetricKeys)));
     }
-
+    Set<String> forbiddenMetrics = new TreeSet<>(metrics.stream()
+      .filter(metric -> ComponentTreeAction.FORBIDDEN_METRIC_TYPES.contains(metric.getValueType()))
+      .map(MetricDto::getKey)
+      .collect(Collectors.toSet()));
+    checkArgument(forbiddenMetrics.isEmpty(), "Metric types '%s' and '%s' are not allowed : %s", DATA.name(), DISTRIB.name(), COMA_JOINER.join(forbiddenMetrics));
```

Comment:
Manually listing the possible values removes the benefit from using FORBIDDEN_METRIC_TYPES. I suggest to simply remove the coupling by changing the label:
``` 
checkArgument(forbiddenMetrics.isEmpty(), "Metrics %s can't be requested in this web service. Please use api/measures/component", COMMA_JOINER.join(forbiddenMetrics));
```

Moreover no need for COMMA_JOINER. You can directly join keys from the stream (`.sorted(...).collect(Collectors.joining(', '))`).